### ISSUE TITLE
`.as_ref().map(|x| *x)` is no longer warned

### ISSUE THREAD

**Issue by y21** [rust-lang/rust-clippy#12142]
### Summary

We used to lint on `Some(1).as_ref().map(|x| *x)`, but no longer do. The interaction between the two lints `useless_asref` and `map_clone` is important here.
Some more context (where this was found and probably for why this is): https://github.com/rust-lang/rust-clippy/pull/12136#discussion_r1451562174.

### Lint Name

`map_clone` / `useless_asref`

### Reproducer

I tried this code:

```rust
fn main() {
  Some(1).as_ref().map(|x| *x);
  Some(1).as_ref().map(|&x| x);
}
```

I expected to see this happen:
```
warning: you are using an explicit closure for copying elements
 --> src/main.rs:2:3
  |
2 |   Some(1).as_ref().map(|x| *x);
  |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider calling the dedicated `copied` method: `Some(1).as_ref().copied()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_clone
  = note: `#[warn(clippy::map_clone)]` on by default

warning: you are using an explicit closure for copying elements
 --> src/main.rs:3:3
  |
3 |   Some(1).as_ref().map(|&x| x);
  |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider calling the dedicated `copied` method: `Some(1).as_ref().copied()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_clone
```

Instead, this happened:
no warnings

### Version

```text
commit a71211d0b52c01f1b37fe544e0e13fd1bdc31979 (Jan 12)
```
